redaxo/src/core/lib/Fragment/Component/Range.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+namespace Redaxo\Core\Fragment\Component;
+
+use Redaxo\Core\Fragment\Fragment;
+use Redaxo\Core\Fragment\HtmlAttributes;
+
+/**
+ * @see redaxo/src/core/fragments/core/Component/Range.php
+ */
+final class Range extends Fragment
+{
+    public function __construct(
+        /**
+         * The range's label.
+         */
+        public string|Fragment|null $label = null,
+
+        /**
+         * Text that describes how to use the range.
+         */
+        public string|Fragment|null $notice = null,
+
+        /**
+         * The name of the range, submitted as a name/value
+         * pair with form data.
+         */
+        public ?string $name = null,
+
+        /**
+         * The current value of the range, submitted as a
+         * name/value pair with form data.
+         */
+        public null|int|float $value = null,
+
+        /**
+         * Disables the range.
+         */
+        public bool $disabled = false,
+
+        /**
+         * The minimum acceptable value of the range.
+         */
+        public null|int|float $min = null,
```

Comment:
psalm meckert, weil ich in den HtmlAttributes (wo diese Property letztlich landet) bisher keine floats zulasse.
floats sind ja tendenziell immer problematisch, mit unerwarteten Rundungen etc.

Weiß nicht, ob ich Dezimalzahlen hier nicht eher als String entgegen nehmen würde. (`'0.1'`).

@staabm hast du da eine Meinung zu?

Reply:
guter punkt. ich denke mit native `string` und via phpdoc eingeschränkt auf `numeric-string` würde es gut funktionieren

Reply:
wobei ich ints schon gerne weiter hier zulassen würde, weil sie wahrscheinlich die häufigst verwendeste variante hier ist. würde also den sperrigen nativen `null|int|string`-typ nehmen, und per phpdoc dann `null|int|numeric-string`.

Reply:
👍